### ISSUE TITLE
llama_get_logits_ith: invalid logits id 14, reason: no logits

### ISSUE THREAD

**Issue by mooreandrew** [ggerganov/llama.cpp#7386]
I am struggling to get embedding models to work, each time reporting the error:

```llama_get_logits_ith: invalid logits id 14, reason: no logits```

I'm using the docker, and have tried this both on Mac and linux. I've tried the following images:

- `ghcr.io/ggerganov/llama.cpp:full--b1-f4dea7d`
- `ghcr.io/ggerganov/llama.cpp:full--b1-854d365`
- `ghcr.io/ggerganov/llama.cpp:server--b1-854d365`
- `ghcr.io/ggerganov/llama.cpp:server`
- `ghcr.io/ggerganov/llama.cpp:full`

Using the following command (for the full image):

```
/bin/bash /app/.devops/tools.s --server -m /models/all-MiniLM-L6-v2.Q4_0.gguf --embeddings --port 8000 --host 0.0.0.0 -n 512
```

I have tried it against the following models and all report the same rrror:

- all-MiniLM-L6-v2.Q4_0.gguf (https://huggingface.co/leliuga/all-MiniLM-L6-v2-GGUF)
- bge-base-en-v1.5-f32.gguf (https://huggingface.co/CompendiumLabs/bge-base-en-v1.5-gguf)

Steps to recreate:

- Start docker image `ghcr.io/ggerganov/llama.cpp:full` with the arguments `--server -m /models/all-MiniLM-L6-v2.Q4_0.gguf --embeddings --port 8000 --host 0.0.0.0 -n 512`
- Open the Server UI (http://localhost:8000) 
- Type anything in the "Say something..." box
- Check the server log and it will show `llama_get_logits_ith: invalid logits id 14, reason: no logits`

I am quite new to this, so it could be something I am doing.

**Comment by ggerganov** [comment]
The Server UI is for chatting. To compute embeddings, you can use for example `curl`:

```bash
curl -X POST "http://localhost:8080/embedding" --data '{"content":"some text to embed"}'
```

**Comment by mooreandrew** [comment]
Ok that makes sense. Thank you!

**Comment by yurivict** [comment]
I am getting this error in ollama while running the inference:
> llama_get_logits_ith: invalid logits id 8, reason: no logits

What might be the cause?
